packages/graphiql/src/YogaGraphiQL.tsx

```diff
@@ -112,7 +112,7 @@ export function YogaGraphiQL(props: YogaGraphiQLProps): React.ReactElement {
       directiveIsRepeatable: true,
       schemaDescription: true,
       ...props,
-      headers: {},
+      headers: JSON.parse(props.headers ?? '{}'),
```

Comment:
props.headers belongs to the state of GraphiQL component that is bound to headers editor. That's why it is string.
So `headers` of GraphiQL props defines the default `headers` value set to the editor component. 
`props.headers` is already passed by fetcher to the executor dynamically with `extensions`.
If you want to have some additional static headers, you should introduce another prop besides `headers`.